src/TDebug.cpp

```diff
@@ -122,9 +156,117 @@ TDebug& TDebug::operator<<(const QList<QString>& list)
 TDebug& TDebug::operator<<(const QList<int>& list)
 {
     for (QList<int>::const_iterator it = list.constBegin(); it != list.constEnd(); ++it) {
-        msg += QString::number(*it);
-        msg += ", ";
+        msg += QStringLiteral("%1, ").arg(QString::number(*it));
     }
     msg += "), ";
     return *this;
 }
+
+void TDebug::changeHostName(const Host* pHost, const QString& newName)
+{
+    if (pHost) {
+        QPair<QString, QChar>& pair = TDebug::smIdentifierMap[pHost];
+        pair.first = newName;
+    }
+}
+
+/* static */ void TDebug::addHost(Host* pHost)
+{
+    if (!pHost) {
+        return;
+    }
+    static const QString validIdentifiers{"123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"};
+    static const int maxIdCount = validIdentifiers.size();
+    int tryCount = -1;
+    do {
+        if (!TDebug::smUsedIdentifiers.contains(validIdentifiers.at(++tryCount))) {
+            break;
+        }
+    } while (tryCount < maxIdCount);
+
+    QString hostName = pHost->getName();
+    if (TDebug::smUsedIdentifiers.contains(validIdentifiers.at(tryCount))) {
+        // Run out of identifers - fall back to '?'
+        QPair<QString, QChar> newIdentifier = qMakePair(hostName, QLatin1Char('?'));
+        TDebug::smIdentifierMap.insert(pHost, newIdentifier);
+    } else {
+        QPair<QString, QChar> newIdentifier = qMakePair(hostName, validIdentifiers.at(tryCount));
+        TDebug::smUsedIdentifiers.insert(validIdentifiers.at(tryCount));
+        TDebug::smIdentifierMap.insert(pHost, newIdentifier);
+    }
+    TDebug localMessage(Qt::blue, Qt::white);
+    localMessage << QStringLiteral("Profile: '%1' started.\n").arg(hostName) >> nullptr;
+    TDebug tableMessage(Qt::white, Qt::black);
+    tableMessage << TDebug::displayNewTable() >> nullptr;
+}
+
+/* static */ void TDebug::removeHost(Host* pHost)
+{
+    TDebug::smIdentifierMap.remove(pHost);
+    TDebug localMessage(Qt::darkGray, Qt::white);
+    localMessage << QStringLiteral("Profile: '%1' ended.\n").arg(pHost->getName()) >> nullptr;
```

Comment:
```suggestion
    localMessage << QStringLiteral("Profile '%1' ended.\n").arg(pHost->getName()) >> nullptr;
```
Remove punctuation in sentence middle

Reply:
As above - so below.